lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -0,0 +1,107 @@
+require 'fastlane/action'
+require 'digest/sha1'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      S3_UPLOADED_FILE_PATH = :S3_UPLOADED_FILE_PATH
+    end
+
+    class UploadToS3Action < Action
+      def self.run(params)
+        file_path = params[:file]
+        file_name = File.basename(file_path)
+        UI.user_error!("Unable to read file at #{file_path}") unless File.file?(file_path)
```

Comment:
You can move this to the `verify_block` of the `ConfigItem` (see below) to let Fastlane report the error earlier in the process while parsing args (without having to create a Runner for nothing)

Reply:
Done in e8748eeb55beb5fd542561b9dc42a5f32e2a7b9e, then extended the pattern to other validation in 684fad43220ea8766b6f7e0f14e1bebacaa4dcf8